src/transforms/lua/v2/interop/log.rs

```diff
@@ -0,0 +1,94 @@
+use crate::event::{Event, LogEvent, Value};
+use rlua::prelude::*;
+
+impl<'a> ToLua<'a> for LogEvent {
+    fn to_lua(self, ctx: LuaContext<'a>) -> LuaResult<LuaValue> {
+        ctx.create_table_from(self.into_iter().map(|(k, v)| (k.to_string(), v)))
+            .map(LuaValue::Table)
+    }
+}
+
+impl<'a> FromLua<'a> for LogEvent {
+    fn from_lua(value: LuaValue<'a>, _: LuaContext<'a>) -> LuaResult<Self> {
+        match value {
+            LuaValue::Table(t) => {
+                let mut log = Event::new_empty_log().into_log();
+                for pair in t.pairs() {
+                    let (key, value): (String, Value) = pair?;
+                    log.insert_flat(key, value);
+                }
+                Ok(log)
+            }
+            _ => Err(rlua::Error::FromLuaConversionError {
+                from: "",
+                to: "LogEvent",
+                message: Some("LogEvent should ba a Lua table".to_string()),
+            }),
+        }
+    }
+}
+
+#[cfg(test)]
+mod test {
+    use super::*;
+
+    #[test]
+    fn to_lua() {
+        let mut event = Event::new_empty_log();
+        let log = event.as_mut_log();
+        log.insert("a", 1);
+        log.insert("nested.field", "2");
+        log.insert("nested.array[0]", "example value");
+        log.insert("nested.array[2]", "another value");
+
+        let assertions = vec![
+            "type(log) == 'table'",
+            "log.a == 1",
+            "type(log.nested) == 'table'",
+            "log.nested.field == '2'",
+            "#log.nested.array == 3",
+            "log.nested.array[1] == 'example value'",
```

Comment:
Oh no, I hate this.

Reply:
In principle, it possible to make it zero-based, as Lua arrays are just hash tables with numeric indexes. But third-party Lua libraries might not handle this well.